### PR TITLE
ledger network update script added

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

ledger network update script added

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

DELETE INCLUSIVE THIS AND BELOW FOR STANDARD PR
------

## Release summary

Version number: [e.g. 1.0.1]

## Release details

Describe in short the main changes with the new release.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side), from `develop`
- [ ] Lint and unit tests pass locally and in CI
- [ ] I have checked the fingerprint hashes are correct by running (`scripts/generate_ipfs_hashes.py`)
- [ ] I have regenerated the latest API docs
- [ ] I built the documentation and updated it with the latest changes
- [ ] I have added an item in `HISTORY.md` for this release
- [ ] I bumped the version number in the `aea/__version__.py` file.
- [ ] I bumped the version number in every Docker image of the repo and published it. Also, I built and published them with tag `latest`  
      (check the READMEs of [`aea-develop`](../develop-image/README.md#publish) 
      and [`aea-user`](../user-image/README.md#publish))
- [ ] I have pushed the latest packages to the registry.
- [ ] I have uploaded the latest `aea` to PyPI.
- [ ] I have uploaded the latest plugins to PyPI.

## Further comments

Write here any other comment about the release, if any.

### CODE REVIEW

**Review by jrriehl — COMMENTED** [review]
Looks good!

File: scripts/ledger_network_update.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+        self.rest_api_address = rest_api_address
+        self.rpc_api_address = rpc_api_address
+
+    def __str__(self) -> str:
+        """Return lines of network configration to be printed."""
+        return (
+            f"Net name: {self.net_name}\n"
+            f"Chain id: {self.chain_id}\n"
+            f"Denom: {self.denom}\n"
+            f"resp api address: {self.rest_api_address}\n"
```
Comment (5A11) [inline scripts/ledger_network_update.py:81]:
```suggestion
            f"REST API address: {self.rest_api_address}\n"
```

File: scripts/ledger_network_update.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+            f"rpc api address: {self.rpc_api_address}\n"
```
Comment (5A11) [inline scripts/ledger_network_update.py:82]:
```suggestion
            f"gRPC address: {self.rpc_api_address}\n"
```

File: scripts/ledger_network_update.py [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+        )
+
+    def get_new_config(self) -> NetworkConfig:
+        """
+        Get new ledger network configuration.
+
+        :return: NetworkConfig instance
+        """
+        net_name = click.prompt("Enter new net name", default=self.cur_config.net_name)
+        chain_id = f"{net_name}-1"
```
Comment (5A11) [inline scripts/ledger_network_update.py:128]: This is not necessarily -1, sometimes another number. Need to get the whole thing via prompt.

File: scripts/ledger_network_update.py [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+        chain_id = click.prompt("Enter new chain_id", default=chain_id)
+        denom = "atestfet"
+        denom = click.prompt("Enter new denom", default=denom)
+        rpc_api_addr = f"https://rpc-{net_name}.fetch.ai:443"
+        rpc_api_addr = click.prompt("Enter new rpc_api_addr", default=rpc_api_addr)
```
Comment (5A11) [inline scripts/ledger_network_update.py:133]:
```suggestion
        rpc_api_addr = click.prompt("Enter the new rpc address", default=rpc_api_addr)
```

File: scripts/ledger_network_update.py [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+        """
+        Get new ledger network configuration.
+
+        :return: NetworkConfig instance
+        """
+        net_name = click.prompt("Enter new net name", default=self.cur_config.net_name)
+        chain_id = f"{net_name}-1"
+        chain_id = click.prompt("Enter new chain_id", default=chain_id)
+        denom = "atestfet"
+        denom = click.prompt("Enter new denom", default=denom)
```
Comment (5A11) [inline scripts/ledger_network_update.py:131]:
```suggestion
        denom = click.prompt("Enter the new denomination", default=denom)
```

File: scripts/ledger_network_update.py [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+        rpc_api_addr = f"https://rpc-{net_name}.fetch.ai:443"
+        rpc_api_addr = click.prompt("Enter new rpc_api_addr", default=rpc_api_addr)
+        rest_api_addr = f"https://rest-{net_name}.fetch.ai:443"
+        rest_api_addr = click.prompt("Enter new rest_api_addr", default=rest_api_addr)
```
Comment (5A11) [inline scripts/ledger_network_update.py:135]:
```suggestion
        rest_api_addr = click.prompt("Enter the new rest api address", default=rest_api_addr)
```

File: scripts/ledger_network_update.py [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+            "Enter cosmpy version (pip style, >=0.2.0)", default=cur_version
+        )
+
+    def run(self):
+        """Do update."""
+        self.cur_config = self.get_current_config()
+        click.echo("Current network config:")
+        click.echo(self.cur_config)
+
+        self.cosmpy_version = self.get_cosmpy_version()
```
Comment (5A11) [inline scripts/ledger_network_update.py:178]: Echo all new information (anything changed). Including new cosmpy version.

File: scripts/ledger_network_update.py [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+                f'FETCHAI_DEFAULT_CHAIN_ID = "{self.new_config.chain_id}"',
+            )
+
+            content = content.replace(
+                f'FETCHAI_DEFAULT_ADDRESS = "{self.cur_config.rest_api_address}"',
+                f'FETCHAI_DEFAULT_ADDRESS = "{self.new_config.rest_api_address}"',
+            )
+
+            content = content.replace(
+                f'FETCHAI_TESTNET_FAUCET_URL = "https://faucet-{self.cur_config.net_name}.t-v2-london-c.fetch-ai.com"',
```
Comment (ianharris-fetch) [inline scripts/ledger_network_update.py:379]: Strictly speaking, this element of the URL is the kubernetes namespace, not the net_name.  So far, all testnets happen to have been deployed into matching namespaces, but that's not necessarily guaranteed (and is guaranteed NOT to be the case for mainnets... not that they have faucets!).

If this update script is targetted at external users, maybe I should create, and this script should reference, faucet-{net_name}.fetch.ai CNAMEs (as we already do for rpc, rest, grpc, explore, browse etc), to better decouple user-facing URLs from internal deployment details?

File: scripts/ledger_network_update.py [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+
+    def update_docs(self):
+        """Update documentation."""
+        click.echo("Update docs")
+        docs_files = (ROOT_DIR / "docs").glob("**/*.md")
+        for f in docs_files:
+            content = f.read_text()
+            content = content.replace(
+                f"https://explore-{self.cur_config.net_name}.fetch.ai",
+                f"https://explore-{self.new_config.net_name}.fetch.ai",
```
Comment (5A11) [inline scripts/ledger_network_update.py:281]: Let's take the whole explorer URL from input.

File: scripts/ledger_network_update.py (hunk 8 above)
Comment (5A11) [reply scripts/ledger_network_update.py:379]: Let's take the whole Faucet URL from input.

File: scripts/ledger_network_update.py (hunk 3 above)
Comment (solarw) [reply scripts/ledger_network_update.py:128]: it's a default value, that will be proposed during the prompt (a line below)

File: scripts/ledger_network_update.py (hunk 9 above)
Comment (solarw) [reply scripts/ledger_network_update.py]: @5A11 done

File: scripts/ledger_network_update.py (hunk 8 above)
Comment (solarw) [reply scripts/ledger_network_update.py]: @5A11 done

**Review by 5A11 — APPROVED** [review]
lgtm

**Pull request merged**